apollo-runtime/src/main/java/com/apollographql/android/cache/normalized/InMemoryNormalizedCacheStore.java

```diff
@@ -0,0 +1,27 @@
+package com.apollographql.android.cache.normalized;
+
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+
+public class InMemoryNormalizedCacheStore extends NormalizedCacheStore {
+
+  private final Map<String, ApolloRecord> store = new HashMap<>();
+
+  @Override public ApolloRecord read(String key) {
+    return store.get(key);
+  }
+
+  @Override public void mergeOrInsert(ApolloRecord apolloRecord) {
+    final ApolloRecord oldRecord = store.get(apolloRecord.key);
+    if (oldRecord == null) {
+      store.put(apolloRecord.key, apolloRecord);
+    } else {
+      oldRecord.mergeWith(apolloRecord);
+    }
+  }
+
+  public Collection<ApolloRecord> getAllRecords() {
```

Comment:
unless there is a `setAllRecords` I don't think the `get` prefix is necessary (similar to how its store.values() not store.getValues())